soccer/Robot.cpp

```diff
@@ -458,7 +458,8 @@ Geometry2d::ShapeSet OurRobot::collectAllObstacles(
     if (_state->ball.valid) {
         // _state->drawShape(ball_obs, Qt::gray,
         //                   QString("ball_obstacles_%1").arg(shell()));
-        fullObstacles.add(createBallObstacle());
+        auto ballObs = createBallObstacle();
+        if (ballObs) fullObstacles.add(ballObs);
```

Comment:
True, although in this case the createBallObstacle() method will only ever return one obstacle, so I don't know if a ShapeSet makes as much sense.